iceoryx_binding_c/CMakeLists.txt

```diff
@@ -43,9 +43,9 @@ iox_add_library(
     BUILD_INTERFACE     ${CMAKE_CURRENT_SOURCE_DIR}/include
                         ${CMAKE_BINARY_DIR}/generated/iceoryx/include
     INSTALL_INTERFACE   include/${PREFIX}
-    PUBLIC_LIBS_LINUX   stdc++ pthread
-    PUBLIC_LIBS_UNIX    stdc++ pthread
-    PUBLIC_LIBS_APPLE   stdc++ pthread
+    PUBLIC_LIBS_LINUX   stdc++
+    PUBLIC_LIBS_UNIX    stdc++
+    PUBLIC_LIBS_APPLE   stdc++
```

Comment:
I think this should be reverted. AFAIK, most of the header in the platform just include the POSIX header which makes pthread a public dependency.